pdns/validate.cc

```diff
@@ -165,6 +165,19 @@ cspmap_t harvestCSPFromRecs(const vector<DNSRecord>& recs)
 
 vState getKeysFor(DNSRecordOracle& dro, const DNSName& zone, keyset_t &keyset)
 {
+  auto luaLocal = g_luaconfs.getLocal();
+
+  /* shortcut to insecure if there is negative trust anchor somewhere up in the
+   * tree
+   */
+  auto negAnchors = luaLocal->negAnchors;
+  if (!negAnchors.empty())
+    for (auto negAnchor : negAnchors)
+      if (zone.isPartOf(negAnchor.first)) {
+        LOG("Found a Negative Trust Anchor for "<<negAnchor.first<<", which was added with the message '"<<negAnchor.second<<"'. Returning Insecure."<<endl);
+        return NTA;
+      }
+
```

Comment:
Shouldn't we check whether there is a (positive) trust anchor below? If I understand correctly section 1.1 of rfc7646, we would be expected to resume validation starting with this trust anchor, right?